slurmpilot/remote_command.py

```diff
@@ -95,6 +97,119 @@ def download_folder(self, remote_path: Path, local_path: Path):
         raise NotImplementedError()
 
 
+class RemoteCommandExecutionSubprocess(RemoteExecution):
+    def __init__(
+        self,
+        master: str,
+        user: str | None = None,
+        proxy: str | None = None,
+        local_dir: str | Path = None,
+    ):
+        """
+        Calls native unix ssh and scp with subprocesses. Should work reliably as long as ssh and scp works in a terminal
+        :param master:
+        :param user:
+        :param proxy:
+        :param local_dir: where logs are written of intermediate commands, if None, use tempdir
+        """
+        super().__init__(master=master, proxy=proxy, user=user)
+        assert (
+            proxy is None
+        ), "not supported"  # would just need to add the proxy in the ssh/scp call
+        self.local_dir = local_dir if local_dir else tempfile.mkdtemp()
+        self.local_dir = Path(self.local_dir)
+
+    def run(
+        self, command: str, pty: bool = False, env: dict | None = None, retries: int = 0
+    ) -> CommandResult:
+        command = f"ssh {self._remote_ssh_hostname()} {command}"
+        return self._run_shell_command(
+            command=command, pty=pty, env=env, retries=retries
+        )
+
+    def _run_shell_command(
+        self, command: str, pty: bool = False, env: dict | None = None, retries: int = 0
+    ) -> CommandResult:
+        # TODO specify dir
+        with open(self.local_dir / "std.out", "w") as stdout:
+            with open(self.local_dir / "std.err", "w") as stderr:
+                process = subprocess.Popen(
```

Comment:
any reason why you don't pass `env` to `Popen`?

Reply:
In this case, we are only calling ssh and scp so I am not sure we need it (and if we need it better not to have it).